python/GangaLHCb/Lib/GaudiRunApp/GaudiRunApp.py

```diff
@@ -0,0 +1,248 @@
+            self.decrementShareCounter(self.is_prepared.name)
+            self.is_prepared = None
+        self.hash = None
+        self.uploadedInput = None
+
+    def prepare(self, force=False):
+        """
+        This method creates a set of prepared files for the application to pass to the RTHandler
+        """
+
+        if (self.is_prepared is not None) and (force is not True):
+            raise ApplicationPrepareError('%s application has already been prepared. Use prepare(force=True) to prepare again.' % getName(self))
+
+        # lets use the same criteria as the configure() method for checking file existence & sanity
+        # this will bail us out of prepare if there's somthing odd with the job config - like the executable
+        # file is unspecified, has a space or is a relative path
+        self.configure(self)
+        logger.info('Preparing %s application.' % getName(self))
+        setattr(self, 'is_prepared', ShareDir())
+        logger.info('Created shared directory: %s' % (self.is_prepared.name))
+
+        build_target = self.buildGangaTarget()
+
+        try:
+            # copy any 'preparable' objects into the shared directory
+            send_to_sharedir = self.copyPreparables()
+            # add the newly created shared directory into the metadata system
+            # if the app is associated with a persisted object
+            self.checkPreparedHasParent(self)
+
+            self.copyIntoPrepDir(build_target)
+            opts_file = self.getOptsFile()
+            if isinstance(opts_file, LocalFile):
+                self.copyIntoPrepDir(path.join( opts_file.localDir, path.basename(opts_file.namePattern) ))
+            elif isinstance(opts_file, DiracFile) and self.master and not isinstance(self.getJobObject().backend, DiracBase):
+                # NB safe to put it here as should have expressly setup a path for this job by now
+                opts_file.get(localPath=self.getSharedPath())
+            else:
+                raise ApplicationConfigurationError("Opts file type %s not yet supported please contact Ganga devs if you require this support" % getName(opts_file))
+            self.post_prepare()
+
+        except Exception as err:
+            logger.debug("Err: %s" % str(err))
+            self.unprepare()
+            raise
+
+        return 1
+
+    def configure(self, masterappconfig):
+        """
+        Required even though nothing is done in this step for this App
+        """
+        # Lets test the inputs
+        opt_file = self.getOptsFile()
+        dir_name = self.getDir()
+        return (None, None)
+
+    def getOptsFile(self):
+        """
+        This function rweturns a sanitized absolute path to the self.myOpts file from user input
+        """
+        if self.myOpts:
+            if isinstance(self.myOpts, LocalFile):
+                ## FIXME LocalFile should return the basename and folder in 2 attibutes so we can piece it together, now it doesn't
```

Comment:
This comment also requires attention. I'd prefer to worry more about the FIXME than the formatting